### PR TITLE
[Feature] Add VinText Converter

### PR DESCRIPTION
**Pull request by Mountchicken**
## Motivation
Add data converter for [VinText](https://github.com/VinAIResearch/dict-guided)

## Modification
add tools for loading and converting labels from .txt annotations
```bash
tools/data/textdet/vintext_converter.py
tools/data/textrecog/vintext_converter.py
```
## Use cases (Optional)
```bash
mkdir vintext && cd vintext
# Download dataset from google drive
wget --load-cookies /tmp/cookies.txt "https://docs.google.com/uc?export=download&confirm=$(wget --quiet --save-cookies /tmp/cookies.txt --keep-session-cookies --no-check-certificate 'https://docs.google.com/uc?export=download&id=1UUQhNvzgpZy7zXBFQp0Qox-BBjunZ0ml' -O- | sed -rn 's/.*confirm=([0-9A-Za-z_]+).*/\1\n/p')&id=1UUQhNvzgpZy7zXBFQp0Qox-BBjunZ0ml" -O vintext.zip && rm -rf /tmp/cookies.txt
# Extract images and annotations
unzip -q vintext.zip && rm vintext.zip
cd vietnamese
# Rename files 
mv labels annotations && mv test_image test && mv train_images  training && mv unseen_test_images  unseen_test
mkdir imgs
mv training imgs/ && mv test imgs/ && mv unseen_test imgs/
# Delete unecessary files
rm general_dict.txt && rm vn_dictionary.txt
# Generate for detection
cd ../../..
python tools/data/textdet/vintext_converter.py data/vintext/vietnamese --nproc 4
# Generate for recognition
python tools/data/textrecog/vintext_converter.py data/vintext/vietnamese --nproc 4
```
WXWorkCapture_16468209109769

### CODE REVIEW

File: tools/data/textdet/vintext_converter.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+            mmcv.imwrite(dst_img, dst_img_path)
+            lines.append(f'{osp.basename(dst_image_root)}/{dst_img_name} '
+                         f'{word}')
+    list_to_file(dst_label_file, lines)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Generate training and test set of FUNSD ')
+    parser.add_argument('root_path', help='Root dir path of FUNSD')
```
Comment (xinke-wang) [inline tools/data/textdet/vintext_converter.py:188]: FUNSD -> VinText

File: tools/data/textdet/vintext_converter.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    args = parse_args()
+    root_path = args.root_path
+    for split in ['training', 'test', 'unseen_test']:
+        print(f'Processing {split} set...')
+        with mmcv.Timer(print_tmpl='It takes {}s to convert FUNSD annotation'):
```
Comment (xinke-wang) [inline tools/data/textdet/vintext_converter.py:204]: FUNSD -> VinText

File: tools/data/textrecog/vintext_converter.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+            mmcv.imwrite(dst_img, dst_img_path)
+            lines.append(f'{osp.basename(dst_image_root)}/{dst_img_name} '
+                         f'{word}')
+    list_to_file(dst_label_file, lines)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Generate training and test set of FUNSD ')
+    parser.add_argument('root_path', help='Root dir path of FUNSD')
```
Comment (xinke-wang) [inline tools/data/textrecog/vintext_converter.py:187]: FUNSD -> VinText

File: tools/data/textrecog/vintext_converter.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    parser.add_argument(
+        '--preserve-vertical',
+        help='Preserve samples containing vertical texts',
+        action='store_true')
+    parser.add_argument(
+        '--filter_nonlatin',
+        help='filter out non-latin instances',
```
Comment (xinke-wang) [inline tools/data/textrecog/vintext_converter.py:194]: filter -> Filter

File: tools/data/textrecog/vintext_converter.py [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+    args = parser.parse_args()
+    return args
+
+
+def main():
+    args = parse_args()
+    root_path = args.root_path
+    for split in ['training', 'test', 'unseen_test']:
+        print(f'Processing {split} set...')
+        with mmcv.Timer(print_tmpl='It takes {}s to convert FUNSD annotation'):
```
Comment (xinke-wang) [inline tools/data/textrecog/vintext_converter.py:207]: FUNSD -> VinText

File: tools/data/textrecog/vintext_converter.py [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+    return images
+
+
+def load_img_info(files):
+    """Load the information of one image.
+
+    Args:
+        files (tuple): The tuple of (img_file, groundtruth_file)
+    Returns:
+        img_info (dict): The dict of the img and annotation information
```
Comment (xinke-wang) [inline tools/data/textrecog/vintext_converter.py:69]: Space line between Returns and Args

File: tools/data/textrecog/vintext_converter.py [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+    return img_info
+
+
+def load_txt_info(gt_file, img_info):
+    """Collect the annotation information.
+
+    Args:
+        gt_file (str): The path to ground-truth
+        img_info (dict): The dict of the img and annotation information
+    Returns:
```
Comment (xinke-wang) [inline tools/data/textrecog/vintext_converter.py:99]: Add space line before Returns

File: tools/data/textdet/vintext_converter.py (hunk 1 above)
Comment (Mountchicken) [reply tools/data/textdet/vintext_converter.py:188]: Ah, I forgot about this. I just copied them from the previous PR😲

File: tools/data/textrecog/vintext_converter.py [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+            assert len(bbox) == 8
+            anno = dict(bbox=bbox, word=word)
+            anno_info.append(anno)
+
+    img_info.update(anno_info=anno_info)
+
+    return img_info
+
+
+def judge_latin(word):
```
Comment (gaotongxiao) [inline tools/data/textrecog/vintext_converter.py:125]: is_latin

**Comment by Mountchicken** [comment]
> TODO:
> 
>     1. Remove the non-Latin character filtering branch as discussed
> 
>     2. Docs


> ## Motivation
> 
> Add data converter for [VinText](https://github.com/VinAIResearch/dict-guided)
> ## Modification
> 
> add tools for loading and converting labels from .txt annotations
> 
> ```shell
> tools/data/textdet/vintext_converter.py
> tools/data/textrecog/vintext_converter.py
> ```
> 
> ### no document updated yet
> ## Use cases (Optional)
> 
> ```shell
> mkdir vintext && cd vintext
> # Download dataset from google drive
> wget --load-cookies /tmp/cookies.txt "https://docs.google.com/uc?export=download&confirm=$(wget --quiet --save-cookies /tmp/cookies.txt --keep-session-cookies --no-check-certificate 'https://docs.google.com/uc?export=download&id=1UUQhNvzgpZy7zXBFQp0Qox-BBjunZ0ml' -O- | sed -rn 's/.*confirm=([0-9A-Za-z_]+).*/\1\n/p')&id=1UUQhNvzgpZy7zXBFQp0Qox-BBjunZ0ml" -O vintext.zip && rm -rf /tmp/cookies.txt
> # Extract images and annotations
> unzip -q vintext.zip && rm vintext.zip
> cd vietnamese
> # Rename files 
> mv labels annotations && mv test_image test && mv train_images  training && mv unseen_test_images  unseen_test
> mkdir imgs
> mv training imgs/ && mv test imgs/ && mv unseen_test imgs/
> # Delete unecessary files
> rm general_dict.txt && rm vn_dictionary.txt
> # Generate for detection
> cd ../../..
> python tools/data/textdet/vintext_converter.py data/vintext/vietnamese --nproc 4
> # Generate for recognition
> python tools/data/textrecog/vintext_converter.py data/vintext/vietnamese --nproc 4
> ```
> 
> WXWorkCapture_16468209109769

**Pull request closed without merging**